m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -565,6 +586,40 @@ public void run() {
             LOG.warn("Processor shut down");
         }
 
+        @Nullable
+        private SizedMetric awaitingPoll() {
```

Comment:
would be nice to have a unit test for this behaviour

Reply:
We do have a unit-test for this